docs/website/docs/glossary.md

```diff
@@ -6,28 +6,114 @@ sidebar_position: 6
+  are the resources, then the source is the spreadsheet. If tables in a database are the resources,
   then the source is the database. A source is organized into one or more resources.
 
+
+
+A dlt Source is the code that handles getting data from an api.
+
+A source contains:
+
+- API methods for authentication, pagination
+- Resources are functions that produce data. Resources are to sources what endpoints are to an API.
+- workflow and customisation code.
+
+A source is the piece of code that produces data. A source might typically be built around a single api. It contains resources, which are data prod
+
+A source contains the **resources**, `source_name().resources` produces a dictionary of resources with the name as key. This allows you to select and execute them individually or in your preferred patterns.
+
+For example, you may want to break down a 50 table copy job into an airflow dag with high parallelism to load the data faster. To do so, you could get the list of resources as
+
+```python
+# get a list of resources' names
+resource_list = sql_source().resources.keys()
+
+#now we are able to make a pipeline for each resource
+for res in resource_list:
+		pipeline.run(sql_source().with_resources(res))
+```
+
+
 ## [Destination](./walkthroughs/create-a-pipeline.md)
 
   The data store where data from the source is loaded (e.g. Google BigQuery).
 
 ## [Pipeline](./walkthroughs/create-a-pipeline.md)
 
-  Moves the data from the source to the destination, according to instructions provided 
+  > **💡** Moves the data from the source to the destination, according to instructions provided
   in the schema (i.e. extracting, normalizing, and loading the data).
 
+
+A pipeline is a connection to the destination. We pass sources or resources to the pipeline. We can also pass generators to the pipeline, . When the pipeline runs, the resources get executed and the data is loaded at destination.
+
+Arguments:
+
+`source` Source may be a dlt source, resource or a generator
+
+`name` is the unique identifier for the pipeline. This unique identifier is used to reference the logs and the state of the pipeline.
+
+`full_refresh` (bool) is a toggle you can use during development to avoid conflicts between pipeline states as you develop them. Full refresh will create a new dataset version with a timestamp in the name. Once you are done developing, make sure to turn it off to funnel the data back into the right dataset.
+
+Example: This pipeline will load the data the generator `gen(10)`produces
+
+```python
+import dlt
+
+def gen(nr):
+    for i in range(nr):
+        yield {'id':1}
+
+pipeline = dlt.pipeline(destination='bigquery',dataset_name='sql_database_data')
+
+info = pipeline.run(gen(10))
+
+print(info)
+```
+
+A pipeline’s unique identifier is the name. If no name is given, then the pipeline takes the name of the source or generator you pass to it.
```

Comment:
it takes the default name from the currently executing module. all the defaults are explained in the general usage documentation and in docstrings